### ISSUE TITLE
Error (ts_gtrends_windows.R#23): `false` must be a `Date` object, not a `Date` object

### ISSUE THREAD

**Issue by uwethuemmel** [trendecon/trendecon#3]
ts_gtrends_windows produces the following error when called as

`ts_gtrends_windows('zara')`

>Error (ts_gtrends_windows.R#23): `false` must be a `Date` object, not a `Date` object

https://github.com/trendecon/trendecon/blob/8da50c5dcf50d9eba9e322c6fc2ef7984610b0ea/R/ts_gtrends_windows.R#L33

**Comment by christophsax** [comment]
For these things, reprexes are very useful: https://reprex.tidyverse.org

**Comment by christophsax** [comment]
I am looking into it, it is weird, though.

**Comment by christophsax** [comment]
Here's the reprex:

``` r
  library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
  dta <- structure(list(start_date = structure(c(17897L, 17904L, 17911L, 17918L, 17925L, 17932L, 17939L, 17946L, 17953L, 17960L, 17967L, 17974L), class = "Date"), end_date = structure(c(17977L, 17984L, 17991L, 17998L, 18005L, 18012L, 18019L, 18026L, 18033L, 18040L, 18047L, 18054L), class = "Date")), class = c("tbl_df", "tbl", "data.frame"), row.names = c(NA, -12L))
  mutate(dta, end_date = if_else(end_date > Sys.Date(), Sys.Date(), end_date))
#> Error: `false` must be a `Date` object, not a `Date` object
```

<sup>Created on 2020-05-21 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by christophsax** [comment]
And the core problem:

``` r

library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
int_date <- tibble(date = structure(c(17977L, 17984L), class = "Date"))
num_date <- tibble(date = as.Date(c("2019-03-22", "2019-03-29")))

mutate(int_date, if_else(date > Sys.Date(), Sys.Date(), date))
#> Error: `false` must be a `Date` object, not a `Date` object
mutate(num_date, if_else(date > Sys.Date(), Sys.Date(), date))
#> # A tibble: 2 x 2
#>   date       `if_else(date > Sys.Date(), Sys.Date(), date)`
#>   <date>     <date>                                        
#> 1 2019-03-22 2019-03-22                                    
#> 2 2019-03-29 2019-03-29
```

<sup>Created on 2020-05-21 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Our `date` is integer after the `rowwise()` `ungroup`, which is not the standard. To make our date standard, we could do:

```r
    rowwise() %>%
    mutate(end_date = seq(start_date, length.out = 2, by = windowsize)[2]) %>%
    ungroup() %>%
    mutate(end_date = as.Date(end_date)) %>%
```

**Comment by uwethuemmel** [comment]
To reproduce the error, run `ts_gtrends_windows('zara')`

The error is generated in line 27 of 00fe4077fcbb11e9d2cf119216682b1c3e70ea82
which is above the suggested fix.

The suggested fix is implemented in 371f98d1ef60a71e897ffc79fff62f69c5f7d2c6 which still produces the error.

The workaround which resolves the error is demonstrated in 05f9a08b9109d38a17a98698f1a396bb36c55fe1

I guess it might really be a bug in dplyr.

**Comment by christophsax** [comment]
Ok, it needs this:

``` r
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union

int_date <- tibble(date = structure(c(17977L, 17984L), class = "Date"))

int_date %>% 
  mutate(date = as.Date(as.numeric(date), origin = "1970-01-01")) %>% 
  mutate(if_else(date > Sys.Date(), Sys.Date(), date))
#> # A tibble: 2 x 2
#>   date       `if_else(date > Sys.Date(), Sys.Date(), date)`
#>   <date>     <date>                                        
#> 1 2019-03-22 2019-03-22                                    
#> 2 2019-03-29 2019-03-29
```

<sup>Created on 2020-05-21 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

We have a corrupt `Date` object. Recreating solves it.

It is not really dplyrs fault that is corrupt. It's our `rowwise() ... ungroup()` magic.
